app/forms/appointment_form.rb

```diff
@@ -0,0 +1,65 @@
+class AppointmentForm
+  include ActiveModel::Model
+
+  BOOKING_REQUEST_ATTRIBUTES = %i(
+    name
+    email
+    phone
+    age_range
+    reference
+    location_name
+    memorable_word
+    accessibility_requirements
+    primary_slot
+    secondary_slot
+    tertiary_slot
+    booking_location
+  ).freeze
+
+  attr_reader :location_aware_booking_request
+
+  attr_accessor :guider_id
+  attr_accessor :location_id
+  attr_accessor :date
+
+  delegate(*BOOKING_REQUEST_ATTRIBUTES, to: :location_aware_booking_request)
+
+  delegate :guiders, to: :booking_location
+
+  def initialize(location_aware_booking_request, params)
+    @location_aware_booking_request = location_aware_booking_request
+    normalise_time(params)
+    super(params)
+  end
+
+  def appointment_params
+    AppointmentMapper.map(self)
+  end
+
+  def flattened_locations
+    FlattenedLocationMapper.map(booking_location)
+  end
+
+  def location_id
+    @location_id ||= location_aware_booking_request.location_id
+  end
+
+  def date
+    @date ||= location_aware_booking_request.primary_slot.date
+  end
+
+  def time
+    @time ||= location_aware_booking_request.primary_slot.delimited_from
+
+    Time.zone.parse("2016-01-01 #{@time}")
```

Comment:
given the date part is ignored can we not just use 
```
 Time.zone.parse("#{@time}")
```
or does it break with daylight saving?